lib/govuk_publishing_components/middleware/ga4_optimise.rb

```diff
@@ -0,0 +1,33 @@
+module GovukPublishingComponents
+  module Middleware
+    class Ga4Optimise
+      DATA_GA4_REGEX = /data-ga4-(?:link|event|form)=("[^"]*")/
```

Comment:
We also have `data-ga4-focus-loss` which takes a JSON. See https://github.com/alphagov/govuk_publishing_components/blob/main/docs/analytics-ga4/trackers/ga4-focus-loss-tracker.md

Reply:
Thanks! Have added focus-loss to the options in the regex.